File: app/src/test/java/com/farmerbb/taskbar/helper/MenuHelperTest.java
package com.farmerbb.taskbar.helper;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

@RunWith(RobolectricTestRunner.class)
public class MenuHelperTest {
    private MenuHelper menuHelper = MenuHelper.getInstance();

    @Test
    public void testGetInstance() {
        assertNotNull(menuHelper);
        for (int i = 1; i <= 20; i++) {
            assertEquals(menuHelper, MenuHelper.getInstance());
        }
    }

    @Test
    public void testSetStartMenuOpen() {
        assertFalse(menuHelper.isStartMenuOpen());
        menuHelper.setStartMenuOpen(true);
        assertTrue(menuHelper.isStartMenuOpen());
        menuHelper.setStartMenuOpen(false);
    }

    @Test
    public void testSetContextMenuOpen() {
        assertFalse(menuHelper.isContextMenuOpen());
        menuHelper.setContextMenuOpen(true);
        assertTrue(menuHelper.isContextMenuOpen());
        menuHelper.setContextMenuOpen(false);
    }
}

File: app/src/test/java/com/farmerbb/taskbar/helper/MenuHelperTest.kt
package com.farmerbb.taskbar.helper

import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class MenuHelperTest {
    private val menuHelper = MenuHelper.getInstance()

    @Test
    fun testGetInstance() {
        Assert.assertNotNull(menuHelper)
        for (i in 1..20) {
            Assert.assertEquals(menuHelper, MenuHelper.getInstance())
        }
    }

    @Test
    fun testSetStartMenuOpen() {
        Assert.assertFalse(menuHelper.isStartMenuOpen)
        menuHelper.isStartMenuOpen = true
        Assert.assertTrue(menuHelper.isStartMenuOpen)
        menuHelper.isStartMenuOpen = false
    }

    @Test
    fun testSetContextMenuOpen() {
        Assert.assertFalse(menuHelper.isContextMenuOpen)
        menuHelper.isContextMenuOpen = true
        Assert.assertTrue(menuHelper.isContextMenuOpen)
        menuHelper.isContextMenuOpen = false
    }
}
